### PR TITLE
fix(neon_talk): Ignore hidden messages for finding previous and last message

### PR DESCRIPTION
**Pull request by provokateurin**
These messages need to be ignored when finding the previous or last message since they won't be displayed in the end.

### CODE REVIEW

File: packages/neon/neon_talk/lib/src/blocs/room.dart [hunk 1]
```diff
@@ -62,7 +62,14 @@ class _TalkRoomBloc extends InteractiveBloc implements TalkRoomBloc {
...
       }
 
-      final lastMessage = result.requireData.firstOrNull;
+      spreed.ChatMessageWithParent? lastMessage;
+      for (final message in result.requireData) {
+        if (!message.isHidden) {
+          lastMessage = message;
+          break;
+        }
+      }
```
Comment (Leptopoda) [inline packages/neon/neon_talk/lib/src/blocs/room.dart:71]: isn't this the same as 
```dart
final lastMessage = result.requireData.firstWhereOrNull((message) => !message.isHidden);
```

I find it easier to understand

File: packages/neon/neon_talk/lib/src/pages/room.dart [hunk 2]
```diff
@@ -119,10 +120,17 @@ class _TalkRoomPageState extends State<TalkRoomPage> {
...
 
                   spreed.ChatMessageWithParent? previousMessage;
-                  if (messagesResult.requireData.length > index + 1) {
-                    previousMessage = messagesResult.requireData[index + 1];
+                  for (var i = index + 1; i < messagesResult.requireData.length; i++) {
+                    final m = messagesResult.requireData[i];
+                    if (!m.isHidden) {
+                      previousMessage = m;
+                      break;
+                    }
```
Comment (Leptopoda) [inline packages/neon/neon_talk/lib/src/pages/room.dart:133]: Again my preference would be:
```dart
final previousMessage =  messagesResult.requireData.skip(index).firstWhereOrNull((message) => !message.isHidden);
```

**Review by Leptopoda — COMMENTED** [review]
I'm wondering why the UI has to skip hidden messages?
This should just be done in the bloc itself (the unwrap callback where the messages subject is populated).

**Comment by provokateurin** [comment]
Ideally I'd like to add tests for the lastKnownMessageId now, but the value would only be used for testing and it should generally be avoided adding code that is only used for testing (and in this case the simple integer would have to be turned into a behaviorsubject which is totally unnecessary).

**Pull request merged**